src/main/java/org/jabref/gui/customizefields/CustomizeGeneralFieldsDialogViewModel.java

```diff
@@ -58,8 +58,9 @@ public void saveFields() {
                 return;
             }
 
-            String testString = CitationKeyGenerator.cleanKey(parts[1], preferences.getUnwantedCharacters());
-            if (!testString.equals(parts[1]) || (parts[1].indexOf('&') >= 0)) {
+            String unwantedChars = "#{}()~,^&-\"'`ʹ\\";
```

Comment:
Please use the unwanted chars from the preferences

Reply:
I thought the problem was that the list of unwanted chars from the preferences did not correspond to the unwanted characters for this particular case. I.e. the characters in the log message below are not the same that are stored in preferences. For example, ';' character is fine in this case because it's used to separate fields, but it is on the list of unwanted characters in preferences

Reply:
Ah okay, thanks for clarifying this.

Reply:
Please also add a changelog entry and fix the checkstyle issue

Reply:
Okay, I'll do later on tonight